server/routes/prisons/configuration/editBookingWindowController.test.ts

```diff
@@ -0,0 +1,141 @@
+
+const prisonService = createMockPrisonService()
+
+const prison = TestData.prison()
+
+beforeEach(() => {
+  flashData = {}
+  flashProvider.mockImplementation(key => flashData[key])
+
+  prisonService.getPrison.mockResolvedValue(prison)
+
+  app = appWithAllRoutes({ services: { prisonService } })
+})
+
+afterEach(() => {
+  jest.resetAllMocks()
+})
+
+describe('Prison booking window edit', () => {
+  const baseUrl = `/prisons/HEI/configuration/booking-window/edit`
+
+  describe('GET /prisons/{:prisonId}/configuration/booking-window/edit', () => {
+    it('should render edit booking window form', () => {
+      prisonService.getPrison.mockResolvedValue(prison)
+
+      return request(app)
+        .get(baseUrl)
+        .expect(200)
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          const $ = cheerio.load(res.text)
+          expect($('h1').text().trim()).toContain('Edit prison booking window')
+          expect($('#policyNoticeDaysMin').val()).toBe('2')
+          expect($('#policyNoticeDaysMax').val()).toBe('28')
+          expect($('[data-test="submit"]').text().trim()).toBe('Update')
+        })
+        .expect(() => {
+          expect(prisonService.getPrison).toHaveBeenCalledTimes(1)
+        })
+    })
+  })
+
+  describe('POST /prisons/{:prisonId}/configuration/booking-window/edit', () => {
+    prisonService.getPrison.mockResolvedValue(prison)
+
+    it('should send valid data to edit booking window and redirect to view template', () => {
+      const updatePrisonDto = TestData.updatePrisonDto({ policyNoticeDaysMin: 10, policyNoticeDaysMax: 20 })
+
+      return request(app)
+        .post(baseUrl)
+        .send(`policyNoticeDaysMin=${updatePrisonDto.policyNoticeDaysMin}`)
+        .send(`policyNoticeDaysMax=${updatePrisonDto.policyNoticeDaysMax}`)
+        .expect(302)
+        .expect('Location', `/prisons/${prison.code}/configuration`)
+        .expect(() => {
+          expect(flashProvider.mock.calls.length).toBe(1)
+          expect(flashProvider).toHaveBeenCalledWith('message', 'Booking window updated')
+          expect(prisonService.updatePrisonDetails).toHaveBeenCalledWith('user1', prison.code, updatePrisonDto)
+        })
+    })
+
+    it('should set validation errors when min and max are below 0 form data', () => {
+      const updatePrisonDto = TestData.updatePrisonDto({ policyNoticeDaysMin: -1, policyNoticeDaysMax: -1 })
+
+      const expectedValidationErrors = [
+        expect.objectContaining({ path: 'policyNoticeDaysMin', msg: 'Enter a min booking window value of at least 1' }),
+        expect.objectContaining({ path: 'policyNoticeDaysMax', msg: 'Enter a max booking window value of at least 1' }),
+      ]
+
+      const expectedFormValues = updatePrisonDto
+
+      return request(app)
+        .post(baseUrl)
+        .send(`policyNoticeDaysMin=${updatePrisonDto.policyNoticeDaysMin}`)
+        .send(`policyNoticeDaysMax=${updatePrisonDto.policyNoticeDaysMax}`)
```

Comment:
Is this the same as doing a single `send` but with a JSON payload of both fields?

Reply:
Good question - I'd just copied the multiple `send.()` for multiple form values because that's what we've done elsewhere 😄. Done some digging in the docs (https://ladjs.github.io/superagent/#post--put-requests) and I _think_ it's correct because sending multiple ones as strings apparently sets `Content-Type` to `application/x-www-form-urlencoded`. Looking in browser requests tab this is how the form is actually submitted. 

Seems like it'll default to `application/json` if you do `.send({ // json object }). Looks like you can use json though if you add a `.type('form')`. Might give that a try next time I need to do this....

Reply:
Update: tested this out and `.type('form')` doesn't seem to make any difference: works fine as separate `.send()` calls with strings or just a JSON one with it all. Will probably use JSON in future as it looks a bit neater and you avoid all the string escaping 👍